collectors/mon.py

```diff
@@ -252,17 +304,108 @@ def _mon_health(self):
+
+        cluster['mon_status'] = mon_status
 
-        cluster['mon_status'] = monstats
+        self.manage_event(health_data.get('overall_status'),
+                          health_data.get('summary', []),
+                          mon_status)
 
         return cluster
 
+    def manage_event(self, health_text, health_summary, mon_status):
+
+        if not self._parent:
+            # invoked without a parent, as part of system tests
+            return
+        elif not self._parent.event_url:
+            # event generation skipped
+            return
+
+        candidates = [mon_name for mon_name in sorted(mon_status)
+                      if mon_status.get(mon_name) == 0]
+
+        if candidates:
+            sender = candidates[0]
+            if sender not in self.ip_names:
+                # only one mon should send, so if that's not us do nothing
+                return
+        else:
+            # no suitable mon to send the alert
+            self.logger.error("Unable to send ANY event - no valid mon "
+                              "found")
+            return
+
+        # If we're here, the current host is suitable to send an event so lets
+        # look deeper to see if we need to
+
+        current_state = CephState(health_text, health_summary)
+        self.logger.debug("health:{}".format(current_state.status))
+        self.logger.debug("health:{}".format(current_state.summary))
+        send_it = False
+
+        if health_text != self.last_state.status:
+            # Overall health has changed, so just send the current state!
+            send_it = True
+        else:
+            # look deeper - only send if the list of issues is different
+            if health_text == 'HEALTH_OK':
+                # nothing to do, nothing to send
+                pass
+            else:
+                if self.last_state.status_items != current_state.status_items:
+                    send_it = True
+
+        self.last_state.update(current_state)
+
+        if send_it:
+            tag = 'health_ok' if current_state.status == 'HEALTH_OK' \
+                else 'health_alert'
+
+            self.logger.info("sending cluster status to "
+                             "{}".format(self._parent.event_url))
+            self.logger.debug(current_state.status_str)
+
+            rc = Mon.post_event(self._parent.event_url,
+                                tag,
+                                current_state.status_str)
+
+            if rc != 200:
+                self.error = True
+                self.error_msgs = ['POST request to {} failed ({}) - '
+                                   'firewall?'.format(self._parent.event_url,
+                                                      rc)]
+                self.logger.warning("Unable to send event - graphite response "
+                                    "{}".format(rc))
+
+        else:
+            # no real change to report
+            pass
```

Comment:
It's a style thing - since it documents the circumstances of how you arrive at that point in the code that's all.